File: ruby/resistor-colors/resistor_colors.rb
module ResistorColors
  ENCODING = %w{
    black
    brown
    red
    orange
    yellow
    green
    blue
    violet
    grey
    white
  }
  
  def self.value(colors)
    colors.map { |color| ENCODING.index(color).to_s }.join.to_i
  end
end

File: haskell/resistor-colors/src/ResistorColors.hs
module ResistorColors (Color(..), value) where

data Color =
    Black
  | Brown
  | Red
  | Orange
  | Yellow
  | Green
  | Blue
  | Violet
  | Grey
  | White
  deriving (Eq, Show, Read, Enum)

value :: [Color] -> Int
value = sum . zipWith evaluateBand [0..] . reverse
  where
    evaluateBand index color = 10 ^ index * fromEnum color
